### ISSUE TITLE
UmbracoModelsBuilder Error (System.Net.Http)

### ISSUE THREAD

**Issue by JaimyTak** [zpqrtbnk/Zbu.ModelsBuilder#221]
Hello!

I have encountered a problem that i'm not able to fix. When I run the Umbraco Models Builder (Builder.cs) I recieve an error message:

UmbracoModelsBuilder: Starting v8.0.5 9/25/2019 9:49:24 AM.
UmbracoModelsBuilder: UmbracoModelsBuilder failed to generate code: FileNotFoundException: Cannot load file or assembly System.Net.Http, Version = 4.2.0.0, Culture = neutral, PublicKeyToken = b03f5f7f11d50a3a or one of its dependencies. The system cannot find the specified file.
UmbracoModelsBuilder: at Umbraco.ModelsBuilder.Api.ApiClient.ValidateClientVersion ()
   at Umbraco.ModelsBuilder.CustomTool.CustomTool.UmbracoModelsBuilder.GenerateRaw (String wszInputFilePath, String wszDefaultNamespace, IntPtr [] rgbOutputFileContents, UInt32 & pcbOutput, String & errMsg.

Can someone please help me?

Thanks in advance.

Kind regards,
Jaimy

**Comment by zpqrtbnk** [comment]
Can you try again with the Visual Studio Extension instead of the Visual Studio CustomTool? See [this post](https://www.zpqrtbnk.net/posts/models-builder-extension/) for details. And then... the error happens right in Visual Studio which is weird - what version of VS is that?

**Comment by JaimyTak** [comment]
Thanks for your reply.

I am recieving the same error with the "Extension". 

My visual studio:

Visual Studio Professional 2017 - Version: 15.9.16

Kind regards,
Jaimy

**Comment by zpqrtbnk** [comment]
I am afraid I haven't tested the latest extension with VS2017  - any chance you can run VS2019?

Else I'll have to troubleshoot.

**Comment by JaimyTak** [comment]
Yes, I'm able to run VS2019. I will send a message/comment when I'm ready/installed everything.

**Comment by JaimyTak** [comment]
I've installed everything.

Extension and CustomTool give both the same errors:

UmbracoModelsBuilder: Starting v8.0.5 9-9-2019 10:57:47.
UmbracoModelsBuilder: UmbracoModelsBuilder failed to generate code: AggregateException: One or more errors occurred.
UmbracoModelsBuilder: at System.Threading.Tasks.Task.ThrowIfExceptional (Boolean includeTaskCanceledExceptions)
   at System.Threading.Tasks.Task`1.GetResultCore (Boolean waitCompletionNotification)
   at System.Threading.Tasks.Task`1.get_Result ()
   at Umbraco.ModelsBuilder.Api.ApiClient.ValidateClientVersion ()
   at Umbraco.ModelsBuilder.CustomTool.CustomTool.UmbracoModelsBuilder.GenerateRaw (String wszInputFilePath, String wszDefaultNamespace, IntPtr [] rgbOutputFileContents, UInt32 & pcbOutput, String & errMsg
UmbracoModelsBuilder: Inner: HttpRequestException: An error occurred while sending the request.
UmbracoModelsBuilder:
UmbracoModelsBuilder: Inner: WebException: The underlying connection was closed: An unexpected error occurred on a send.
UmbracoModelsBuilder: at System.Net.HttpWebRequest.EndGetRequestStream (IAsyncResult asyncResult, TransportContext & context)
   at System.Net.Http.HttpClientHandler.GetRequestStreamCallback (IAsyncResult ar)
UmbracoModelsBuilder: Inner: IOException: Unable to read data from the transport connection: The external host has broken a connection.
UmbracoModelsBuilder: at System.Net.TlsStream.EndWrite (IAsyncResult asyncResult)
   at System.Net.PooledStream.EndWrite (IAsyncResult asyncResult)
   at System.Net.ConnectStream.WriteHeadersCallback (IAsyncResult ar)
UmbracoModelsBuilder: Inner: SocketException: The external host has disconnected
UmbracoModelsBuilder: at System.Net.Sockets.Socket.EndReceive (IAsyncResult asyncResult)
   at System.Net.Sockets.NetworkStream.EndRead (IAsyncResult asyncResult)
UmbracoModelsBuilder: AggregateInner: HttpRequestException: An error occurred while sending the request.
UmbracoModelsBuilder:
